### ISSUE TITLE
Not receiving firebase push notifications

### ISSUE THREAD

**Issue by kvp18** [onnela-lab/beiwe-backend#225]
I have a study design I am testing internally prior to enrolling participants. Unfortunately, I cannot get the push notifications to work. The study will use only iOS devices. I have tried re-creating and re-uploading the Firebase credentials to the study server and have not had any luck. I also followed the troubleshooting docs: https://github.com/onnela-lab/beiwe-backend/wiki/Celery-troubleshooting but was unsuccessful. Any help/advice would be greatly appreciated.

Below is a summary of the current statuses I see and checks I have run:
- I ssh into the working and see the following:
  - ![CleanShot 2021-08-18 at 12 37 05](https://user-images.githubusercontent.com/62187430/129937430-e8679bb4-4272-4587-aeb6-0a643e8ae41a.png)
- When I run `nano /var/mail/ubuntu` to see `mail` I get:
  - ![CleanShot 2021-08-18 at 12 38 42](https://user-images.githubusercontent.com/62187430/129937641-92e35098-26c8-4144-ae65-1fd57f857980.png)
- No errors in `~/celery_processing.log` or `~/celery_forest.log`
- Failed attempts in `~/celery_push_send.log`:
  - ![CleanShot 2021-08-18 at 12 40 57](https://user-images.githubusercontent.com/62187430/129938019-03f4edf9-2cd6-4c1c-bb90-fafb039063ee.png)
- Celery appears to be running when checking via `htop` 
- Checked archived events in the RDS and see the following: `Auth error from APNS or Web Push Service`

**Comment by zagorsky** [comment]
`Auth error from APNS or Web Push Service` is the problem.  Everything else is working OK.  There's some issue with the credentials Firebase has, where they're not working with Apple Push Notification Services.  Sorry, I'm not sure exactly what this problem is, although I've seen it before.

**Comment by biblicabeebli** [comment]
For all inquiries about push notifications not working please get in direct contact with me at biblicabeiwe at gmail dot com to request working push notification credentials.

(I'm still reviewing issues, will have more public-facing updates soon, I can only type things so quickly...)

**Comment by biblicabeebli** [comment]
closing this issue as it should be resolved.  If not feel free to comment and we will reopen.

**Comment by er42** [comment]
Sorry to reopen an old closed thread, but I think I might be having a related issue. I've been trying to get firebase notifications to work without much success...

On iOS devices, all attempted notifications result in `Auth error from apns or web push service`, and no survey notifications appear on the device. There is also a slightly more informative error on Sentry: `401 Client Error: Unauthorized for url: https://fcm.googleapis.com/v1/projects/[OURPROJECTNAME]/messages:send`

On Android devices, survey notifications do show up on the device, but they seem to be triggered locally (?) as I am getting a different error for the push notifications in the Beiwe logs: `SenderId mismatch`

I generated all the firebase credentials using the directions exactly as described in https://github.com/onnela-lab/beiwe-backend/wiki/Firebase-Configuration, and I've triple-checked that the credential files are correct and were uploaded to the server successfully. I also went through all the troubleshooting steps on https://github.com/onnela-lab/beiwe-backend/wiki/Celery-troubleshooting, and that didn't fix things, nor did it lead to any other clues apart from those above.

After googling these errors, my hunch is that there is some issue with our firebase project/beiwe backend not being properly authorized to push notifications to the Beiwe2 apps. Is there something else we have to do to request/enable this? For example, do we need to provide an APN key/certificate on the Firebase project page? (we do not have one, currently)
